Libraries/Text/TextInput/RCTBaseTextInputView.m

```diff
@@ -475,7 +475,7 @@ - (BOOL)textInputShouldChangeTextInRange:(NSRange)range replacementText:(NSStrin
     }
   }
 
-  NSString *previousText = backedTextInputView.attributedText.string ?: @"";
+  NSString *previousText = [backedTextInputView.attributedText.string copy] ?: @""; // TODO(OSS Candidate ISS#2710739)
```

Comment:
Great find!

The `-[NSAttributedString string]` property is marked `readonly, copy` but it appears that Apple deliberately ignores it per their documentation: https://developer.apple.com/documentation/foundation/nsattributedstring/1412616-string?language=objc

> For performance reasons, this property returns the current backing store of the attributed string object. If you want to maintain a snapshot of this as you manipulate the returned string, you should make a copy of the appropriate substring

Since this string is being processed by the background JS thread it is in danger of being mutated by the foreground thread.   In this case, based on your XCode screenshot, it's crashing in `convertIdToFollyDynamic` presumably when it calls `     NSData *data = [json dataUsingEncoding:NSUTF8StringEncoding];`.   So the background thread is encoding a string and the foreground thread changes it before it completes.

We should do the same manual `copy` on all the places where the string is being passed to the JS thread.   I just audited the source and found these places were we should also be copying the `attributedText.string` before queuing it to the JS thread:
`textInputDidBeginEditing`
`textInputDidEndEditing`
`textInputShouldReturn`
`textInputDidChange`

And the same fix should be contributed upstream.

Reply:
Also, minor nit: we should either use all dot notation, or all conventional notation.   for the macOS changes we prefer conventional notation, but upstream uses mostly dot notation.   If this goes upstream, they may prefer all dot notation.

Reply:
Regarding the nit: I'm under the impression that the dot notation is only used for properties. Since `copy` is a function, I think it's correct to the use conventional notation here.

Reply:
Of course.   One of the reasons I hate dot notation.  I confuses me 🤪

Reply:
Upstream: https://github.com/facebook/react-native/pull/29307